### PR TITLE
Feat adding sms adapters

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

It added support for the following SMS providers (Adapters)

- Seven (formerly sms77)
- Sinch 
- Infobip
- Clickatell
- Plivo

Plus I've added one document summarizing and comparing the prices for a single unit and 10K for the implemented providers.

## Test Plan

I've run tests on each provider. Plivo I had for a long time and for everyone except Clickatell has a trial.

## Related PRs and Issues

#17 

### Have you read the [Contributing Guidelines on issues]
Yes

### CODE REVIEW

**Comment by stnguyen90** [comment]
@byawitz, thanks for the PR! 🙏🏼 it looks like there are some linter problems. Would you please address those? You should be able to run `composer format` to fix all the lint problems.

**Comment by byawitz** [comment]
@stnguyen90 Thanks for your answer

Done

File: src/Utopia/Messaging/Adapters/SMS/Clickatell.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// Reference Material
+// https://docs.clickatell.com/channels/sms-channels/sms-api-reference/#tag/SMS-API/operation/sendMessageREST_1
+class Clickatell extends SMSAdapter
+{
+    /**
+     * @param  string  $apiKey Clickatell API Key
+     */
+    public function __construct(
+  private string $apiKey,
+ ) {
```
Comment (abnegate) [inline src/Utopia/Messaging/Adapters/SMS/Clickatell.php:17]: Looks like the indenting is a little off in all the adapters, not sure why the linter isn't picking this up, please indent all the constructor parameters with 8 spaces :pray:

File: src/Utopia/Messaging/Adapters/SMS/Clickatell.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Clickatell';
+    }
+
+    public function getMaxMessagesPerRequest(): int
+    {
+        return 1000;
```
Comment (abnegate) [inline src/Utopia/Messaging/Adapters/SMS/Clickatell.php:27]: I notice the maximum is set to 1000 for all the new adapters, please double check the max values are correct for all services

**Review by abnegate — COMMENTED** [review]
Looks good! A couple of minor comments. Please address, as well as add some screenshots of successful sends using these adapters :pray:

File: src/Utopia/Messaging/Adapters/SMS/Clickatell.php (hunk 2 above)
Comment (byawitz) [reply src/Utopia/Messaging/Adapters/SMS/Clickatell.php]: I was able to verify as follow:
**Plivo** - up to 1000
https://www.plivo.com/docs/sms/api/message#bulk-messaging

**Clickatell** - up to 500
https://6194477.fs1.hubspotusercontent-na1.net/hubfs/6194477/SMS%20REST%20API%20v2.0-1.pdf

**Sinch** - up to 1000
https://developers.sinch.com/docs/sms/api-reference/sms/tag/Batches/

About Seven & Infobin there isn't any information about there _limits_ so I left it on 1,000

**Comment by byawitz** [comment]
image

I don't know what image I should attach for the verification. Hope this one helps.

Also, @abnegate I've sent you a DM with the test suite code that includes all the API keys for you to test.

**Review by abnegate — APPROVED** [review]

**Pull request merged**